priam/src/main/java/com/netflix/priam/cluster/management/Compaction.java

```diff
@@ -0,0 +1,169 @@
+
+            if (cassandraOperations != null)
+                cassandraOperations.getKeyspaces().forEach(keyspaceName -> {
+                List<String> existingCfs = columnFamilyFilter.getOrDefault(keyspaceName, new ArrayList<>());
+                columnFamilyFilter.put(keyspaceName, existingCfs);
+            });
+            return columnFamilyFilter;
+        }
+
+        logger.info("Compaction: Override provided by user. Calculating KS/CF's to qualify for compaction");
+        String[] filters = config.getCompactionCFList().split(",");
+        for (int i = 0; i < filters.length; i++) { //process each filter
+            if (columnFamilyFilterPattern.matcher(filters[i]).find()) {
+
+                String[] filter = filters[i].split("\\.");
+                String keyspaceName = filter[0];
+                String columnFamilyName = filter[1];
+
+                if (columnFamilyName.indexOf("-") != -1)
+                    columnFamilyName = columnFamilyName.substring(0, columnFamilyName.indexOf("-"));
+
+                logger.info("Compaction: Adding CF: {}.{}", keyspaceName, columnFamilyName);
+
+                List<String> existingCfs = columnFamilyFilter.getOrDefault(keyspaceName, new ArrayList<>());
+                if (!columnFamilyName.equalsIgnoreCase("*"))
+                    existingCfs.add(columnFamilyName);
+                columnFamilyFilter.put(keyspaceName, existingCfs);
+
+            } else {
+                throw new IllegalArgumentException("Column family filter format is not valid.  Format needs to be \"keyspace.columnfamily\".  Invalid input: " + filters[i]);
+            }
+        }
+
+        return columnFamilyFilter;
+    }
+
+    /*
+     * @return the keyspace(s) compacted.  List can be empty but never null.
+     */
+    protected String runTask() throws Exception {
+        List<String> compactionKeyspaces = new ArrayList<String>();
+        final Map<String, List<String>> columnFamilyFilter = updateCompactionCFList(config);
+
+        if (columnFamilyFilter == null || columnFamilyFilter.isEmpty()) {
+            logger.warn("No op on requested \"compaction\" as there are no keyspaces.");
+            return compactionKeyspaces.toString();
+        }
+
+        for (Map.Entry<String, List<String>> entry : columnFamilyFilter.entrySet()) {
+            String keyspace = entry.getKey();
+            List<String> columnfamilies = entry.getValue();
+
+            if (!cassandraOperations.getKeyspaces().contains(keyspace)) {
+                logger.error("Keyspace: {} configured for compaction does not exist!", keyspace);
+                continue;
+            }
+
+            if (SchemaConstant.isSystemKeyspace(keyspace)) //no need to compact system keyspaces.
+                continue;
+
+            try {
+                if (columnfamilies == null || columnfamilies.isEmpty())
+                    cassandraOperations.forceKeyspaceCompaction(keyspace,  null);
```

Comment:
When would you get into this code path, considering the updateCompactionCFList logic above?

Reply:
If there are no filters provided, we just empty arraylist for the CF's. Later, we call compact on each keyspace (so we can avoid system keyspace). 
```List<String> existingCfs = columnFamilyFilter.getOrDefault(keyspaceName, new ArrayList<>());```

Reply:
Did you test passing `null` would not break anything?

Reply:
Did you test when passing `null` to forceKeyspaceCompaction?

Reply:
Yes. CassandraOperations usually take care of 'null' inputs.